tests/lib/rules/require-expect.js

```diff
@@ -0,0 +1,195 @@
+            options: []
+        },
+
+        // CallExpression with no object doesn't error
+        {
+            code: "test('name', function(assert) { assert.expect(0); other() });",
+            options: []
+        },
+
+        // assert at top of test context is ok
+        {
+            code: "test('name', function(assert) { assert.ok(true) });",
+            options: ["except-simple"]
+        },
+
+        // assert with expect at the top of test context is ok
+        {
+            code: "test('name', function(assert) { assert.expect(0); if (false) { assert.ok(false) } });",
+            options: ["except-simple"]
+        },
+
+        // nested modules
+        {
+            code: [
+                "module('a', function() {",
+                "    test('a - test', function(assert) {",
+                "        assert.ok(true, 'no expect needed');",
+                "    });",
+                "",
+                "    module('b', function() {",
+                "        test('b - test', function(assert) {",
+                "            assert.ok(true, 'still no expect needed');",
+                "        });",
+                "    });",
+                "});"
+            ].join(returnAndIndent),
+            options: ["except-simple"]
+        }
+
+    ],
+
+    invalid: [
+        // default, make sure expect is identified correctly
+        {
+            code: "test('name', function(assert) { other.assert.expect(0) });",
+            errors: [{
+                message: EXCEPT_SIMPLE_ERROR_MESSAGE,
+                type: "CallExpression"
+            }],
+            options: []
+        },
+
+        // assert in loop block
+        {
+            code: "test('name', function(assert) { while (false) { assert.ok(true) } });",
+            errors: [{
+                message: ALWAYS_ERROR_MESSAGE,
+                type: "CallExpression"
+            }],
+            options: ["except-simple"]
+        },
+
+        // assert used in callback
+        {
+            code: "test('name', function(assert) { maybe(function() { assert.ok(true) }); });",
+            errors: [{
+                message: ALWAYS_ERROR_MESSAGE,
+                type: "CallExpression"
+            }],
+            options: ["except-simple"]
+        },
+
+        // assert in function expression
+        {
+            code: "test('name', function(assert) { var maybe = function() { assert.ok(true) }; });",
+            errors: [{
+                message: ALWAYS_ERROR_MESSAGE,
+                type: "CallExpression"
+            }],
+            options: ["except-simple"]
+        },
+
+        // `expect` does not count when used inside of a block.
+        {
+            code: "test('name', function(assert) { function name() { assert.expect(0); assert.ok(true) } });",
+            errors: [{
+                message: ALWAYS_ERROR_MESSAGE,
+                type: "CallExpression"
+            }],
+            options: ["except-simple"]
+        },
```

Comment:
Nice test case. Can you add a similar one for the callback case (expect and assertion both in callback, test should be flagged)?